src/components/Home/Highlights/PoliCarousel.tsx

```diff
@@ -0,0 +1,170 @@
+import { PaginationCarousel } from "./PaginationCarousel"
+import lecturesImage from "assets/carousel-lectures.png"
+import iseeImage from "assets/carousel-isee.png"
+import { CarouselItem } from "./HighlightsManager"
+
+const { width } = Dimensions.get("window")
+
+/**
+ * enum to differentiate the different types of widget we could have
+ * different widget types have different background images
+ */
+export enum WidgetType {
+    LECTURES,
+    ISEE,
+}
+
+/**
+ * custom Carousel component, it receives the data to show from the HighlightsManager
+ */
+export const PoliCarousel: FC<{ dataToShow: CarouselItem[] }> = ({
+    dataToShow,
+}) => {
+    const [currentIndex, setCurrentIndex] = useState<number>(0)
+    const scrollX = useRef(new Animated.Value(0)).current
+
+    //questa parte funziona ma non va bene scritta cosi
+    const handleViewableItemsChanged = useRef(
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ({ changed }: { changed: any }) => {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            if (changed[0].isViewable) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+                setCurrentIndex(changed[0].index)
+            }
+        }
+    ).current
+
+    const viewAbilityConfig = { viewAreaCoveragePercentThreshold: 50 }
+
+    return (
+        <View style={{ marginTop: 60 }}>
+            <Animated.FlatList
+                data={dataToShow}
+                keyExtractor={(_, index) => index.toString()}
+                bounces={true}
+                pagingEnabled={true}
+                horizontal
+                showsHorizontalScrollIndicator={false}
+                onScroll={Animated.event(
+                    [
+                        {
+                            nativeEvent: {
+                                contentOffset: { x: scrollX },
+                            },
+                        },
+                    ],
+                    { useNativeDriver: false }
+                )}
+                renderItem={({ item }) => {
+                    return (
+                        <View>
+                            <Pressable
+                                style={{
+                                    width,
+                                    justifyContent: "center",
+                                    alignItems: "center",
+                                }}
+                                onPress={() =>
+                                    console.log("Highlights premuto!")
+                                }
+                            >
+                                <ImageBackground
+                                    style={{
+                                        width: width - 80,
+                                        height: 77,
+                                        borderRadius: 10,
```

Comment:
quasi dimenticavo, su iOS questa cosa non funziona come dovrebbe, devi mettere overflow hidden perché l'immagine abbia i bordi arrotondati
```suggestion
                                        borderRadius: 10,
                                        overflow: "hidden",
```

Reply:
confermo!Ora funziona e risolve anche il problema che avevo prima e che ti ho fatto vedere su telegram